tower-reconnect/src/lib.rs

```diff
@@ -23,47 +24,47 @@ pub enum Error<T, U> {
     NotReady,
 }
 
-pub struct ResponseFuture<T, Request>
+pub struct ResponseFuture<T, Ctx, Request>
 where
     // TODO:
     // This struct should just be generic over the response future, but
     // doing so would require changing the future's error type
-    T: NewService<Request>,
+    T: MakeService<Ctx, Request>,
 {
     inner: Option<<T::Service as Service<Request>>::Future>,
 }
 
 #[derive(Debug)]
-enum State<T, Request>
-where
-    T: NewService<Request>
-{
+enum State<F, S> {
     Idle,
-    Connecting(T::Future),
-    Connected(T::Service),
+    Connecting(F),
+    Connected(S),
 }
 
 // ===== impl Reconnect =====
 
-impl<T, Request> Reconnect<T, Request>
+impl<T, Ctx, Request> Reconnect<T, Ctx, Request>
 where
-    T: NewService<Request>,
+    T: MakeService<Ctx, Request>,
+    Ctx: Clone,
```

Comment:
I'm thinking that instead of requiring `Clone` here, it'd be more flexible if the bounds were `T: MakeService<&Ctx, Request>`. Then, if the new `Service` needs a clone, it can make one, but it's otherwise not wasted.

Reply:
How did that pan out?